### PR TITLE
Feature/editor photos

### PR DESCRIPTION
**Pull request by Oksana-Z**

### CODE REVIEW

**Comment by Oksana-Z** [comment]
code:
-logout
-changed profile.js(vuex)
-implemented editing images (vue-cropper.js)
-created component EditImage.vue
-added notifications in Edit profile page
-added styles, quasar componenents in Edit profile page
-Edit profile page communicates with backend throw Vuex

**Comment by igorkokotko** [comment]
@Oksana-Z please resolve conflicts

File: client/src/views/Settings/Tabs/EditProfile/index.vue [hunk 1]
```diff
@@ -1,140 +1,251 @@
...
-        v-model="profile.socialAccounts.facebook"
-        type="url"
-      />
+
+      <q-card class="my-card">
+        <q-card-section>
+          <div class="text-h6">Avatar</div>
+        </q-card-section>
+        <q-card-section>
+          <template v-if="photoUrl===''">
```
Comment (igorkokotko) [inline client/src/views/Settings/Tabs/EditProfile/index.vue]: I'm a bit confused why you check if photoUrl equals to empty string and inside that block you pass to avatar component :img  which is empty string.

File: client/src/views/Settings/Tabs/EditProfile/index.vue [hunk 2]
```diff
@@ -1,140 +1,251 @@
...
+          </div>
+        </q-card-section>
+      </q-card>
+
+      <q-card class="my-card">
+        <q-card-section>
+          <div class="text-h6">Cover photo</div>
+        </q-card-section>
+        <q-card-section>
+          <template v-if="backgroundUrl===''">
```
Comment (igorkokotko) [inline client/src/views/Settings/Tabs/EditProfile/index.vue]: The same here, please explain why do you really need such weird behavior?

File: client/src/views/Settings/Tabs/EditProfile/index.vue [hunk 3]
```diff
@@ -1,140 +1,251 @@
...
+                src="@/assets/img/cover_photo.jpg"
+                class="coverPhoto"
+              />
+            </div>
+          </template>
+          <template v-else>
+            <edit-image
+              class="backgroundImage"
+              :src="backgroundUrl"
+              :aspectRatio="16 / 9"
```
Comment (igorkokotko) [inline client/src/views/Settings/Tabs/EditProfile/index.vue]: Don't use magic numbers. Store them in data.

File: client/src/views/Settings/Tabs/EditProfile/index.vue [hunk 4]
```diff
@@ -1,140 +1,251 @@
...
+        actions: [{ icon: 'close', color: 'white' }],
+        timeout: 3000
       }
     }
   },
 
   computed: {
     profileGetter () {
-      return this.$store.getters.profileGetter
+      return this.$store.getters['profile/myProfile']
```
Comment (igorkokotko) [inline client/src/views/Settings/Tabs/EditProfile/index.vue]: Please use getters like this -  https://vuex.vuejs.org/ru/guide/getters.html#%D0%B2%D1%81%D0%BF%D0%BE%D0%BC%D0%BE%D0%B3%D0%B0%D1%82%D0%B5n%D1%8C%D0%BD%D0%B0%D1%8F-%D1%84%D1%83%D0%BD%D0%BA%D1%86%D0%B8%D1%8F-mapgetters

File: client/src/views/Settings/Tabs/EditProfile/index.vue [hunk 5]
```diff
@@ -145,47 +256,115 @@ export default {
   },
 
   created () {
-    this.$store.dispatch('uploadProfile')
+    this.$store.dispatch('profile/getMyProfile')
```
Comment (igorkokotko) [inline client/src/views/Settings/Tabs/EditProfile/index.vue]: Use mapActions.

File: client/src/views/Settings/Tabs/EditProfile/index.vue [hunk 6]
```diff
@@ -145,47 +256,115 @@ export default {
...
   },
 
   methods: {
-    createValue (val, done) {
-      if (val.length > 2) {
-        if (!suggestedInterests.includes(val)) {
-          done(val, 'add-unique')
-        }
+    uploadPhotoUrl (val) {
+      if (val.target.files[0].size > 2097152) {
```
Comment (igorkokotko) [inline client/src/views/Settings/Tabs/EditProfile/index.vue]: Don't use magic numbers. Declare intuitive variable why you need exactly this number value.

File: client/src/views/Settings/Tabs/EditProfile/index.vue [hunk 7]
```diff
@@ -145,47 +256,115 @@ export default {
...
-      this.photoUrl = URL.createObjectURL(val[0])
+    deletePhotoUrl (val) {
+      this.photoUrl = ""
+      this.profile.avatar_url = ""
     },
 
     uploadBackgroundUrl (val) {
-      this.profile.user_background = val[0]
-      this.backgroundUrl = URL.createObjectURL(val[0])
+      if (val.target.files[0].size > 2097152) {
```
Comment (igorkokotko) [inline client/src/views/Settings/Tabs/EditProfile/index.vue]: You have almost identical method - uploadPhotoUrl, does it make sense to reuse it?

File: client/src/App.vue [hunk 8]
```diff
@@ -26,6 +27,19 @@ export default {
   data () {
     return {
     }
+  },
+
+  created () {
+    if (window.localStorage.getItem('token') && window.localStorage.getItem('token') !== '') {
```
Comment (igorkokotko) [inline client/src/App.vue]: Please reuse - https://github.com/igorkokotko/emOcean/blob/develop/client/src/views/Authentication/Authorized.js#L2

**Review by igorkokotko — CHANGES_REQUESTED** [review]

**Review by igorkokotko — APPROVED** [review]

**Pull request merged**